src/client/types.ts

```diff
@@ -140,6 +140,11 @@ export interface CreateTaskOptions {
    */
   scheduledTime?: Date;
 
+  /**
+   * The end time for a task. If not defined the task will never expire
```

Comment:
this is really when the _schedule_ for the task ends, not an actual run of a task, right? if so, i'd update the documentation to mention that and rename it to `scheduleEndTime` or something similar to indicate that this is about scheduling, not stopping a currently running task

Reply:
This is more like the task will not execute a run past this time. It will be applicable to scheduledStartTime as well. For instance if the scheduledStartTime is 5 days from now but the endTime is 4 days from now the nextRunTime will be undefined and the task would be marked as failed.

Reply:
Renaming to lastRunTime